File: fsharp/armstrong-numbers/ArmstrongNumbers.fs
module ArmstrongNumbers

open System

let isArmstrongNumber (number: int): bool =
    let digits = number.ToString()
    let len = digits.Length

    digits
    |> Seq.map(fun x -> pown (Char.GetNumericValue x) len)
    |> Seq.sum
    |> (=) number

File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        var ary = number.ToString().ToCharArray();
        return number == (int)ary.Select(n => Math.Pow(char.GetNumericValue(n), ary.Length)).Sum();
    }
}
